backend/engine/plugins/gitlab_repo_health/README.md

```diff
@@ -0,0 +1,310 @@
+    "subrules": {
+        "any_of": [
+            {
+                "type": "composite_rule",
+                "id": "repo_ghas_secret_scanning_and_code_scanning",
+                "subrules": {
+                    "all_of": [
+                        {
+                            "type": "repo_secret_scanning",
+                            "require_push_protection": true
+                        },
+                        {
+                            "type": "repo_code_scanning"
+                        }
+                    ]
+                }
+            },
+            {
+                "type": "composite_rule",
+                "id": "branch_checks_secrets_and_static_analysis",
+                "subrules": {
+                    "any_of": [
+                        {
+                            "type": "branch_protection_status_checks",
+                            "checks": {
+                                "all_of": [
+                                    "secrets",
+                                    "static-analysis"
+                                ]
+                            }
+                        },
+                        {
+                            "type": "branch_rule_status_checks",
+                            "checks": {
+                                "all_of": [
+                                    "secrets",
+                                    "static-analysis"
+                                ]
+                            }
+                        }
+                    ]
+                }
+            }
+        ]
+    }
+}
+```
+
+### Branch Protection Commit Signing
+
+For checking that a branch protection rule is enabled that enforces commit signing.
+
+#### Example <!-- omit from toc -->
+
+```json
+{
+    "type": "branch_protection_commit_signing"
+}
+```
+
+### Branch Protection Enforce Admins
+
+Checks that a branch protection rule is enabled that enforces branch protection
+for admins. Mapped to "Do not allow bypassing the above settings" in branch
+protection rules in Github's UI
+
+#### Example <!-- omit from toc -->
+
+```json
+{
+    "type": "branch_protection_enforce_admins"
+}
+```
+
+### Branch Protection Pull Requests
+
+For checking that a branch protection rule is enabled that requires pull requests.
+
+#### Fields
+
+##### `min_approvals`
+
+Minimum number of approvals that must be configured. This should be a number
+
+##### `expect`
+
+Fields to expect in the response. Refer to fields within the
+"required_pull_request_reviews" object in [Github's Get Branch Protection API
+Endpoint](https://docs.github.com/en/rest/branches/branch-protection#get-branch-protection).
+
+#### Example <!-- omit from toc -->
+
+```json
+{
+    "type": "branch_protection_pull_requests",
+    "min_approvals": 1,
+    "expect": {
+        "dismiss_stale_reviews": true,
+        "require_code_owner_reviews": true,
```

Comment:
Are these fields the same in Gitlab? Same applies to other `expect` fields

Reply:
It seems they are different, so these should be updated (like they are in the default config) https://github.com/WarnerMedia/artemis/pull/243/files/ec5e89710e8041a3860eaf40553e89b3672651c6#diff-1cfd707557e630a0ded8379d2ec834087029d58c64959d787fd618733b2626e5R42